backend/src/lib/clusters.ts

```diff
@@ -0,0 +1,19 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { IResource } from '../resources/resource'
+import { getKubeResources } from '../routes/events'
+
+export type ClusterMapType = {
+  [key: string]: IResource
+}
+export function getClusterMap(): ClusterMapType {
+  const managedClusters = getKubeResources('ManagedCluster', 'cluster.open-cluster-management.io/v1')
+  const clusterDeployments = getKubeResources('ClusterDeployment', 'hive.openshift.io/v1')
+  const hostedClusters = getKubeResources('HostedCluster', 'hypershift.openshift.io/v1beta1')
+  return [...managedClusters, ...clusterDeployments, ...hostedClusters].reduce((clusterMap, cluster) => {
+    if (cluster.metadata.name) {
+      clusterMap[cluster.metadata.name] = cluster
+    }
+    return clusterMap
+  }, {} as ClusterMapType)
+}
```

Comment:
Only clusters that are under management have search running on them. I'm not sure why we would need to include `ClusterDeployment` or `HostedCluster` resources when assembling the list of cluster names. Is there something I'm missing?